lib/actions/user.js

```diff
@@ -444,3 +444,20 @@ export function saveUserPlace (placeToSave, placeIndex) {
     dispatch(createOrUpdateUser(clonedUser, true))
   }
 }
+
+/**
+ * Delete the place data at the specified index for the logged-in user.
+ */
+export function deleteUserPlace (placeIndex) {
+  return function (dispatch, getState) {
+    // Get the places as shown (and not as retrieved from db), so that the index passed from URL applies
+    // (indexes 0 and 1 are for 'Home' and 'Work' places).
```

Comment:
If a user has home saved and an "other" place, this statement about reserved indexes doesn't necessarily hold true, but I'm splitting hairs! I think the way you've stated this is OK.

Reply:
Maybe it makes sense to do that. Let me try that out.

Reply:
Good call on this. https://github.com/opentripplanner/otp-react-redux/pull/280/commits/d121350ed90b22cfbfea83e84aa256be5e5b4e6f has the location sorting done upfront and simplifies the rest of the code a bit.

Reply:
When I said we should pre-sort on the persistence/API side, I more so meant in otp-middleware, rather than in the UI.

Reply:
Displaying Home and Work on top is solely a UI feature, so it makes most sense to process the locations on the UI side rather than the backend.